File: py/ini_reader.py
from constants import LOGGER


class IniReader:
    def __init__(self, file_path: str) -> None:
        self.data: dict[str, dict[str, str]] = {}
        self._read_data(file_path)

    def _read_data(self, file_path: str) -> None:
        with open(file_path, "r") as ini:
            lines = ini.readlines()

        section = ""

        for full_line in lines:
            line = full_line.split(";", 1)[0].strip()  # Remove comments
            if len(line) == 0:
                continue

            if line[0] == "[" and line[-1] == "]":
                # new section
                section = line[1:-1]
                continue

            if "=" not in line:
                LOGGER.warning(f"Invalid line: {full_line}")

            key, value = line.split("=", 1)

            self.data[section] = self.data.get(section, {}) | {
                key.strip(): value.strip()
            }

    def get(self, key: str) -> dict[str, str]:
        return self.data.get(key)

    def get(self, key: str, default: dict[str, str]) -> dict[str, str]:
        return self.data.get(key, default)

    def __str__(self) -> str:
        return str(self.data)


if __name__ == "__main__":
    # Test out INI file reading
    reader = IniReader("data/rooms.ini")

    import json

    LOGGER.info(json.dumps(reader.data, indent=4))

File: src/ini_reader.js
import { displaySeed } from "./script.js";

class IniReader {
  data;

  /**
   * @param {string} file_path
   */
  constructor(file_path) {
    this.data = {};
    this.#_readData(file_path);
  }

  #_readData(file_path) {
    fetch(file_path)
      .then((res) => res.text())
      .then((text) => {
        let lines = text.split("\n").map((line) => line.trim());

        let section = "";
        for (let full_line of lines) {
          let line = full_line.split(";", 1)[0].trim(); // Remove comments
          if (line.length == 0) continue;

          if (line[0] == "[" && line[line.length - 1] == "]") {
            section = line.substring(1, line.length - 1);
            continue;
          }

          if (line.indexOf("=") == -1) {
            console.warn(`Invalid line: ${full_line}`);
            continue;
          }
          let [key, value] = line.split("=", 2);
          this.data[section] = this.data[section] || {};
          this.data[section][key.trim()] = value.trim();
        }
        displaySeed();
      });
  }

  get(key) {
    return this.data[key];
  }

  get(key, default_value) {
    return this.data[key] ?? default_value;
  }
}

export { IniReader };
